pkg/allocator/allocator.go

```diff
@@ -241,6 +241,18 @@ type Backend interface {
 	// releases.Release(ctx context.Context, key AllocatorKey) (err error)
 	Release(ctx context.Context, id idpool.ID, key AllocatorKey) (err error)
 
+	// UpdateMasterKey refreshes the master key part of an identity.
+	// When reliablyMissing is set it will also recreate missing master key.
+	UpdateMasterKey(ctx context.Context, id idpool.ID, key AllocatorKey, reliablyMissing bool) error
+
+	// ValidateSlaveKey validates that the slave part of the record that this node is using this key -> id
+	// mapping is up-to-date. If it returns true, it means that no action is needed.
+	ValidateSlaveKey(ctx context.Context, id idpool.ID, key AllocatorKey) (bool, error)
+
+	// UpdateSlaveKey refreshes the slave part of the record that this node is using this key -> id
+	// mapping. When reliablyMissing is set it will also recreate missing slave key.
+	UpdateSlaveKey(ctx context.Context, id idpool.ID, key AllocatorKey, reliablyMissing bool) error
```

Comment:
Trying to keep the existing lingua around these. In a follow-up PR we might want to change these to `upsertX` instead, as that is in practice what the implementation does (at lease kvstore), and then only use `reliablyMissing` for the initial operation.